airgun/entities/host_new.py

```diff
@@ -159,6 +160,23 @@ def get_ansible_roles_modal(self, entity_name):
         self.browser.plugin.ensure_page_safe()
         return view.table.read()
 
+    def enable_tracer(self, entity_name):
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        view.traces.enable_traces.click()
+        modal = EnableTracerView(self.browser)
+        if modal.is_displayed:
```

Comment:
Mostly following convention in the file for dealing with modals - there isn't a meaningful situation where the modal doesn't pop up, the test should just fail at that point.

Reply:
Shouldn't an exception be thrown in that case? Or just handle it somehow, don't leave it in undefined state unless something, somewhere fails?

Reply:
I could throw an exception in that case explicitly. It will also always fail if this modal doesn't pop up since the next assertion won't be correct if the modal dialogue isn't interacted with. That's required for the title on the page to change.

Reply:
Generally speaking, I don't think this is necessary here.

Reply:
`It will also always fail if this modal doesn't pop up since the next assertion won't be correct`
In your specific test, yes. In any other use, who knows.